tests/test_pointcloud/test_pointcloud.py

```diff
@@ -1 +1,238 @@
+
+        # 3/ Check implicit loading when calling a function requiring .ds
+        pc = PointCloud(self.fn_las, data_column="Z")
+        assert not pc.is_loaded
+
+        pc.buffer(distance=0.1)
+        assert pc.is_loaded
+
+    def test_load__errors(self) -> None:
+        """Test errors raised by loading."""
+
+        pc = PointCloud(self.fn_las, data_column="Z")
+        pc.load()
+
+        # Error if already loaded
+        with pytest.raises(ValueError, match="Data are already loaded."):
+            pc.load()
+
+        pc = PointCloud(self.fn_las, data_column="Z")
+        pc._name = None
+        with pytest.raises(AttributeError, match="Cannot load as filename is not set anymore.*"):
+            pc.load()
+
+    def test_data_column(self) -> None:
+        """Test the setting and getting of the main data column."""
+
+        # Assert column is set properly at instantiation
+        pc = PointCloud(self.gdf1, data_column="b1")
+        assert pc.data_column == "b1"
+
+        # And can be reset to another name if it exists
+        pc2 = PointCloud(self.gdf2, data_column="b1")
+        assert pc2.data_column == "b1"
+        # First syntax
+        pc2.data_column = "b2"
+        assert pc2.data_column == "b2"
+        # Equivalent syntax
+        pc2.set_data_column("b1")
+        assert pc2.data_column == "b1"
+
+    def test_data_column__errors(self) -> None:
+        """Test errors raised during setting of data column."""
+
+        pc = PointCloud(self.gdf1, data_column="b1")
+        # If the data column does not exist
+        with pytest.raises(ValueError, match="Data column column_that_does_not_exist not found*"):
+            pc.data_column = "column_that_does_not_exist"
+        # Equivalent syntax
+        with pytest.raises(ValueError, match="Data column column_that_does_not_exist not found*"):
+            pc.set_data_column("column_that_does_not_exist")
+
+    def test_pointcloud_equal(self) -> None:
+        """Test pointcloud equality."""
+
+    def test_from_array(self) -> None:
+        """Test building point cloud from array."""
+
+        # Build from array and compare
+        pc1 = PointCloud(self.gdf1, data_column="b1")
+        pc_from_arr = PointCloud.from_array(array=self.arr_points, crs=4326, data_column="b1")
+        assert pc_from_arr.pointcloud_equal(pc1)
+
+        # Should be the same with transposed array
+        pc_from_arr = PointCloud.from_array(array=self.arr_points.T, crs=4326, data_column="b1")
+        assert pc_from_arr.pointcloud_equal(pc1)
+
+    def test_from_array__errors(self) -> None:
+        """Test errors raised during creation with array."""
+
+        array = np.ones((4, 5))
+        with pytest.raises(ValueError, match="Array must be of shape 3xN or Nx3."):
+            PointCloud.from_array(array=array, crs=4326)
+
+    def test_from_tuples(self) -> None:
+        """Test building point cloud from list of tuples."""
+
+        pc1 = PointCloud(self.gdf1, data_column="b1")
```

Comment:
the fact that you have to specify the data column each time makes me think. By default, if there is only one column, should it not pick that one?